src/utils.ts

```diff
@@ -109,6 +109,26 @@ export function joinURL (base: string, ...input: string[]): string {
   return url
 }
 
+export function withHttp (input: string): string {
+  return withProtocol(input, 'http:')
+}
+
+export function withHttps (input: string): string {
+  return withProtocol(input, 'https:')
+}
+
+export function withoutProtocol (input: string): string {
+  const parsed = parseURL(input)
+  parsed.protocol = ''
+  return stringifyParsedURL(parsed, true)
+}
+
+export function withProtocol (input: string, protocol?: string): string {
+  const parsed = parseURL(input)
+  parsed.protocol = protocol
+  return stringifyParsedURL(parsed)
```

Comment:
Similarly we could use a regex to detect protocol and either directly prepend if there is no protocol already or replace regex match position without expensive parse/stringify